src/bridges/uniswap/UniswapBridge.sol

```diff
@@ -0,0 +1,504 @@
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        bool inputIsEth = _inputAssetA.assetType == AztecTypes.AztecAssetType.ETH;
+        bool outputIsEth = _outputAssetA.assetType == AztecTypes.AztecAssetType.ETH;
+
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20 && !inputIsEth) {
+            revert ErrorLib.InvalidInputA();
+        }
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20 && !outputIsEth) {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        Path memory path = _decodePath(
+            inputIsEth ? WETH : _inputAssetA.erc20Address,
+            _auxData,
+            outputIsEth ? WETH : _outputAssetA.erc20Address
+        );
+
+        uint256 inputValueSplitPath1 = (_inputValue * path.percentage1) / 100;
+
+        if (path.percentage1 != 0) {
+            // Swap using the first swap path
+            outputValueA = UNI_ROUTER.exactInput{value: inputIsEth ? inputValueSplitPath1 : 0}(
+                ISwapRouter.ExactInputParams({
+                    path: path.splitPath1,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountIn: inputValueSplitPath1,
+                    amountOutMinimum: 0
+                })
+            );
+        }
+
+        if (path.percentage2 != 0) {
+            // Swap using the second swap path
+            uint256 inputValueSplitPath2 = _inputValue - inputValueSplitPath1;
+            outputValueA += UNI_ROUTER.exactInput{value: inputIsEth ? inputValueSplitPath2 : 0}(
+                ISwapRouter.ExactInputParams({
+                    path: path.splitPath2,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountIn: inputValueSplitPath2,
+                    amountOutMinimum: 0
+                })
+            );
+        }
+
+        uint256 tokenInDecimals = 18;
+        if (!inputIsEth) {
+            try IERC20Metadata(_inputAssetA.erc20Address).decimals() returns (uint8 decimals) {
+                tokenInDecimals = decimals;
+            } catch (bytes memory) {
+                emit DefaultDecimalsWarning();
+            }
+        }
+        uint256 amountOutMinimum = (_inputValue * path.minPrice) / 10**tokenInDecimals;
+        if (outputValueA < amountOutMinimum) revert InsufficientAmountOut();
+
+        if (outputIsEth) {
+            IWETH(WETH).withdraw(outputValueA);
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice A function which encodes min price to the format used in this bridge.
+     * @param _minPrice - Minimum acceptable price multiplied by 10**tokenInDecimals
```

Comment:
Also added a comment in one of the test related to this.

Reply:
The initial motivation for for passing in _minPrice multiplied by 10**tokenInDecimals was that in cases where decimals of token in are higher than decimals of tokenOut computing _minPrice would get rounded to 0 (amountIn/amountOut would get rounded to 0). Multiplying amountIn by 10**tokenInDecimals before dividing it with amountOut solved this issue.

I acknowledge that it's a bad design decision since it's definitely something which should stay in the internal logic and not be exposed to contract users.

I addressed this issue in e7b217c967569085d6bedbf36ed47a3fb7d2d143 by completely changing the interface of the function. I think now it's pretty clear to use and it avoids the issue mentioned above.